Fails to build on 32-bit arch

https://github.com/tpm2-software/tpm2-tools/pull/1829/files#diff-735b9b0b3bafb20ec3b9294457d73f65R168

```
lib/tpm2_util.c: In function ‘tpm2_util_string_to_int32’:
lib/tpm2_util.c:168:22: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
     if (errno || tmp > UINT32_MAX) {
                      ^
```

This is with `gcc (Raspbian 6.3.0-18+rpi1+deb9u1) 6.3.0 20170516` on a Raspberry Pi running Debian 9 armhf.

`signed long int tmp` is 32 bits long and can't exceed a `UINT32_MAX`. So there isn't a logic error in the comparison, but the warning plus the `-Werror` config do make the build fail.

Affects master and 4.1-rc0.

--- a/lib/tpm2_util.c
+++ b/lib/tpm2_util.c
@@ -165,7 +165,7 @@ bool tpm2_util_string_to_int32(const char *str, int32_t *value) {
     /* clear errno before the call, should be 0 afterwards */
     errno = 0;
     signed long int tmp = strtol(str, &endptr, 0);
-    if (errno || tmp > UINT32_MAX) {
+    if (errno || tmp > INT32_MAX) {
         return false;
     }
 

